### PR TITLE
Switch out the switch

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Switch out the switch

Before
----------------------------------------
Q When is a switch not a switch?

image

After
----------------------------------------
A. When it can be replaced with a single check....

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Contribute/Import/Parser/Contribution.php [hunk 1]
```diff
@@ -586,60 +586,50 @@ public function &getImportedContributions() {
...
 
+    if (!empty($params['pledge_id'])) {
+      if (CRM_Core_DAO::getFieldValue('CRM_Pledge_DAO_Pledge', $params['pledge_id'], 'contact_id') != $contributionContactID) {
+        throw new CRM_Core_Exception('Invalid Pledge ID provided. Contribution row was skipped.', CRM_Import_Parser::ERROR);
       }
+      $values['pledge_id'] = $params['pledge_id'];
     }
+
+    // we need to check if oldest payment amount equal to contribution amount
+    require_once 'CRM/Pledge/BAO/PledgePayment.php';
```
Comment (seamuslee001) [inline CRM/Contribute/Import/Parser/Contribution.php:623]: @eileenmcnaughton is the require once necessary?

Comment (eileenmcnaughton) [reply CRM/Contribute/Import/Parser/Contribution.php:623]: @seamuslee001 nope - definitely not - I just didn't touch it in this change to keep it 'clean' - given all the whitespace change

Comment (colemanw) [reply CRM/Contribute/Import/Parser/Contribution.php:623]: I say delete it

Comment (eileenmcnaughton) [reply CRM/Contribute/Import/Parser/Contribution.php]: ok- I added a commit to rip them out - I think the whole function can be migrated to apiv4 - ie

add `pledge_id` as a pseudo-param on `Contribution::create`

**Pull request merged**